lib/src/geo/crs/crs.dart

```diff
@@ -304,17 +299,19 @@ class Proj4Crs extends Crs {
   num zoom(double scale) {
     // Find closest number in _scales, down
     var downScale = _closestElement(_scales, scale);
+    if (downScale == null) {
+      return double.negativeInfinity;
+    }
     var downZoom = _scales.indexOf(downScale);
     // Check if scale is downScale => return array index
     if (scale == downScale) {
       return downZoom;
     }
-    if (downScale == null) {
-      return double.negativeInfinity;
-    }
     // Interpolate
     var nextZoom = downZoom + 1;
     var nextScale = _scales[nextZoom];
```

Comment:
Is there any cases where `nextZoom` can be out of `_scales` boundaries ?

Reply:
Honestly I'm not deep enough into this part to tell. I deleted the three lines above.